utils/aeternity.js

```diff
@@ -100,25 +143,98 @@ class Aeternity {
       : fetchOracleState();
   }
 
+  addAdditionalTipsData(tips) {
+    return tips.map(data => {
+      const tip = data;
+      tip.topics = [...new Set(tip.title.match(topicsRegex))].map(x => x.toLowerCase());
+
+      tip.amount_ae = Util.atomsToAe(tip.amount).toFixed();
+      tip.total_amount_ae = Util.atomsToAe(tip.total_amount).toFixed();
+      tip.total_unclaimed_amount_ae = Util.atomsToAe(tip.total_unclaimed_amount).toFixed();
+      tip.total_claimed_amount_ae = Util.atomsToAe(tip.total_claimed_amount).toFixed();
+
+      tip.retips = tip.retips.map(retipData => {
+        const retip = retipData;
+        retip.amount_ae = Util.atomsToAe(retip.amount).toFixed();
+        return retip;
+      });
+
+      return tip;
+    });
+  }
+
   async getTips() {
     if (!this.client) throw new Error('Init sdk first');
     const fetchTips = async () => {
       const state = await this.contract.methods.get_state();
-      return this.getTipsRetips(state.decodedResult);
+      const { tips } = tippingContractUtil.getTipsRetips(state.decodedResult);
+      return this.addAdditionalTipsData(tips);
     };
 
     return this.cache
       ? this.cache.getOrSet(['getTips'], () => fetchTips(), this.cache.shortCacheTime)
       : fetchTips();
   }
 
-  getContractSource() {
-    if (!process.env.CONTRACT_FILE) throw new Error(`env.CONTRACT_FILE is ${process.env.CONTRACT_FILE}`);
-    return fs.readFileSync(`${__dirname}/${process.env.CONTRACT_FILE}.aes`, 'utf-8');
+  async getTokenRegistryState() {
+    const fetchData = async () => this.tokenRegistry.methods.get_state().then(r => r.decodedResult);
+
+    return this.cache
```

Comment:
are we testing the behavior of aeternity.js without cache anywhere? Is it realistic that the backend runs without cache?

Reply:
right, this is legacy from when cache was first introduced and not mandatory to be initialized. I'll remove the checks